packages/odyssey-react-mui/src/OdysseyThemeProvider.tsx

```diff
@@ -10,17 +10,71 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
+import React, {
+  createContext,
+  ReactNode,
+  useCallback,
+  useContext,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
 import {
   createTheme,
   ThemeProvider as MuiThemeProvider,
 } from "@mui/material/styles";
-import { memo, ReactNode, useMemo } from "react";
 
 import { ThemeOptions } from "@mui/material";
 import { deepmerge } from "@mui/utils";
 import { createOdysseyMuiTheme, DesignTokensOverride } from "./theme";
 import * as Tokens from "@okta/odyssey-design-tokens";
 import { OdysseyDesignTokensContext } from "./OdysseyDesignTokensContext";
+import { ContrastMode, getBackgroundColor } from "./ThemeContrastProvider";
+
+const useContrastMode = (explicitContrastMode?: ContrastMode) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [contrastMode, setContrastMode] = useState<ContrastMode>(
+    () => explicitContrastMode || "highContrast",
+  );
+
+  const updateContrastMode = useCallback(() => {
+    if (!explicitContrastMode) {
+      const newBgColor = getBackgroundColor(ref.current);
+      setContrastMode(
+        newBgColor === Tokens.HueNeutral50 ? "highContrast" : "lowContrast",
+      );
+    }
+  }, [explicitContrastMode]);
```

Comment:
We should be calling these `backgroundColor` rather than `bgColor`.

And in this case, I don't think we need to say `newBgColor` either as that's confusing. That's the current background color.

```suggestion
  const updateContrastMode = useCallback(() => {
    if (!explicitContrastMode) {
      const backgroundColor = getBackgroundColor(ref.current);
      setContrastMode(
        backgroundColor === Tokens.HueNeutral50 ? "highContrast" : "lowContrast",
      );
    }
  }, [explicitContrastMode]);
```

Reply:
Also, does this hook need to be here in the first place? We shouldn't be checking for `Tokens.HueNeutral50` outside of `ContrastModeProvider`. I think we can probably remove this altogether or use the condition from another hook.